nav2_costmap_2d/include/nav2_costmap_2d/footprint_subscriber.hpp

```diff
@@ -20,43 +20,27 @@
 #include "rclcpp/rclcpp.hpp"
 #include "nav2_costmap_2d/footprint.hpp"
 #include "nav2_util/lifecycle_node.hpp"
+#include "nav2_util/simple_subscriber.hpp"
 
 namespace nav2_costmap_2d
 {
 
-class FootprintSubscriber
+class FootprintSubscriber : public nav2_util::SimpleSubscriber<geometry_msgs::msg::PolygonStamped>
 {
 public:
   FootprintSubscriber(
```

Comment:
FYI: In ROS2, the pattern of which constructors to provide isn't well established yet. However, the prevalent pattern seems to be providing the interface-based constructor and the rclcpp::Node-based constructor. They seem to be providing lifecycle support by requiring a lifecycle node to use the interface pointers. See tf2_ros/message_filter.h, for example. 

There are a few options for us to choose from:

1. Separate constructors for: rclcpp::Node, rclcpp_lifecycle::LifecycleNode, node interfaces (all three options)
2. Separate constructors for: rclcpp::Node, node interfaces (this is what many ROS2 classes are using)
3. Separate constructors for: rclcpp:Node, rclcpp_lifecycle::LifecycleNode (what's used here)

Previously, Carlos used #1 (with the old robot class). For consistency, we should standardize on an approach when implementing helper/support classes that work on behalf of a node.